src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -1100,6 +1100,82 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/snapshots/{id}/authDomain:
+    get:
+      tags:
+        - snapshots
+        - repository
+      description: Retrieve the auth domain for the snapshot
+      operationId: getSnapshotAuthDomain
+      parameters:
+        - $ref: '#/components/parameters/Id'
+      responses:
+        200:
+          description: Auth domain
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+        400:
+          description: Bad request - invalid id, badly formed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+        403:
+          description: No permission to see auth domain
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+        404:
+          description: Not found - snapshot id does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+    patch:
+      tags:
+        - snapshots
+        - repository
+      description: Patch the auth domain for the snapshot
```

Comment:
Can you add something in the description that auth domain groups (or whatever we want to call these...we should double check) can only be added but never removed?